hibernate-core/src/main/java/org/hibernate/dialect/CockroachDialect.java

```diff
@@ -111,8 +115,42 @@ public CockroachDialect() {
 	}
 
 	public CockroachDialect(DialectResolutionInfo info) {
-		super(info);
-		driverKind = PostgreSQLDriverKind.determineKind( info );
+		this( createVersion( info ), PostgreSQLDriverKind.determineKind( info ) );
+		registerKeywords( info );
+	}
+
+	protected static DatabaseVersion createVersion( DialectResolutionInfo info ) {
+		DatabaseVersion databaseVersion = null;
+
+		if ( info.getDatabaseMetadata() != null ) {
+			try (java.sql.Statement s = info.getDatabaseMetadata().getConnection().createStatement() ) {
+				final ResultSet rs = s.executeQuery( "SELECT version()" );
+				if ( rs.next() ) {
+					databaseVersion = parseVersion( rs.getString( 1 ) );
+				}
+			}
+			catch (SQLException ex) {
+				// Ignore
+			}
+		}
+		if ( databaseVersion == null ) {
+			throw new HibernateException("Could not determine the CockroachDB version");
+		}
+		return databaseVersion;
```

Comment:
Just like we don't throw an exception in `Dialect.checkVersion()` we also shouldn't throw here. Please adapt that to return the minimum version and emit a warning log message.

```suggestion
		if ( databaseVersion == null ) {
			// todo: log a warning that the version can't be determined
			return MINIMUM_VERSION;
		}
		return databaseVersion;
```